packages/sirius-web/backend/sirius-web-papaya/src/main/java/org/eclipse/sirius/web/papaya/representations/table/ProjectTableRepresentationDescriptionProvider.java

```diff
@@ -113,10 +121,16 @@ private List<Object> getAllContents(EObject object) {
         return result;
     }
 
-    private List<Object> getSemanticElements(VariableManager variableManager) {
-        return variableManager.get(VariableManager.SELF, ProjectSpec.class)
+    private PaginatedData getSemanticElements(VariableManager variableManager) {
+        var optionalPagination = variableManager.get(TableRenderer.PAGINATION, CursorBasedPaginationData.class);
+        var optionalEditingContext = variableManager.get(IEditingContext.EDITING_CONTEXT, IEditingContext.class);
+        var allRows = variableManager.get(VariableManager.SELF, PackageSpec.class)
```

Comment:
The goal is to stop asking for all the contents. I don't have any proof that this PR is working has expected given the state of those tests.